Modules/socketmodule.c

```diff
@@ -6396,11 +6396,19 @@ Convert a 32-bit integer from network to host byte order.");
 
 
 static PyObject *
-socket_htons(PyObject *self, PyObject *args)
```

Comment:
````
/*[clinic input]
_socket.socket.htons
    x: int
    /

Convert a 16-bit unsigned integer from host to network byte order.
[clinic start generated code]*/
````

Use AC instead of writing manual parsing.
I got a similar performance gain in my local environment.

```
AS-IS: 5000000 loops, best of 5: 60.9 nsec per loop
TO-BE: 10000000 loops, best of 5: 23.3 nsec per loop
```

Reply:
@wrongnull 
You should take a look at the https://devguide.python.org/development-tools/clinic/ if you're not familiar with Argument Clinic :)

Reply:
I'm pretty sure a similar approach can be applied to all other functions with only one argument. If yes, I'll be happy to make such a commit in this branch.

Reply:
@corona10 seems ready. clinic argument was also applied to other functions where it was beneficial

Reply:
I'm not sure if ValueError is more preferable if negative value is passed to ntohl and htonl functions

Reply:
> I'm not sure if ValueError is more preferable if negative value is passed to ntohl and htonl functions

Please don't change the exception type; it will occur in user regression.
Can I know where the type of exception will be changed if AC is applied?

Reply:
> > I'm not sure if ValueError is more preferable if negative value is passed to ntohl and htonl functions
> 
> Please don't change the exception type; it will occur in user regression. Can I know where the type of exception will be changed if AC is applied?

Exception type will change to ```ValueError``` from ```OverflowError``` in ```htonl``` and ```ntohl``` functions after applying ```usigned_long``` argument converter.

Reply:
> Exception type will change to ValueError from OverflowError in htonl and ntohl functions after applying usigned_long argument converter.

In that case, I would like to recommend not to touch it.

Reply:
> > Exception type will change to ValueError from OverflowError in htonl and ntohl functions after applying usigned_long argument converter.
> 
> In that case, I would like to recommend not to touch it.

OK, I reverted this changes